### ISSUE TITLE
`aws ecr get-login-password` fails when using `--profile`

### ISSUE THREAD

**Issue by paulroujansky** [aws/aws-cli#8202]
### Describe the bug

Authenticating to an ECR registry as explained in https://docs.aws.amazon.com/AmazonECR/latest/userguide/getting-started-cli.html fails when passing `--profile` and therefore assuming a role in a sub-account rather than logging through it directly.

This worked until recently.

### Expected Behavior

```shell
$ aws ecr get-login-password --region eu-west-3 --profile myprofile | docker login --username AWS --password-stdin 763104351884.dkr.ecr.eu-west-3.amazonaws.com
Login Succeeded
```

### Current Behavior

```shell
$ aws ecr get-login-password --region eu-west-3 --profile myprofile | docker login --username AWS --password-stdin 763104351884.dkr.ecr.eu-west-3.amazonaws.com

'f61ea6b82400b5502489dbfc01aa219feb7e0391'
Error: Cannot perform an interactive login from a non TTY device
```

### Reproduction Steps

Login into the root account and assuming role on sub-account:
```shell
aws ecr get-login-password --region eu-west-3 --profile myprofile | docker login --username AWS --password-stdin 763104351884.dkr.ecr.eu-west-3.amazonaws.com
```

"myprofile" points to the sub-account where I have "AdministratorAccess".
.aws/credentials contains the `aws_access_key_id` and `aws_secret_access_key` of the root account.
.aws/config contains the following info:

```
[profile myprofile]
region = eu-west-3
role_arn = arn:aws:iam::MYACCOUNTID:role/role-ops
source_profile = default
```

### Possible Solution

Login directly into the sub-account:
```shell
aws ecr get-login-password --region eu-west-3 | docker login --username AWS --password-stdin 763104351884.dkr.ecr.eu-west-3.amazonaws.com
```

.aws/credentials contains the `aws_access_key_id` and `aws_secret_access_key` of the sub-account ("myprofile").

### Additional Information/Context

Docker version 24.0.6, build ed223bc

### CLI version used

aws-cli/2.13.21 Python/3.11.5 Windows/10 exe/AMD64 prompt/off

### Environment details (OS name and version, etc.)

Windows 10

**Comment by tim-finnigan** [comment]
Hi @paulroujansky thanks for reaching out. You mentioned that this worked until recently – have you made any changes to your account prior to this happening? I see that you're using the latest version of the CLI, but I don't think there are any recent ECR changes in the [CHANGELOG](https://github.com/aws/aws-cli/blob/v2/CHANGELOG.rst?plain=1) that would cause that error. 

The error `Cannot perform an interactive login from a non TTY device` has been reported several times in [GitHub](https://github.com/aws/aws-cli/issues?q=is%3Aissue+%22Cannot+perform+an+interactive+login+from+a+non+TTY+device%22+is%3Aclosed) and [Stack Overflow](https://www.google.com/search?client=firefox-b-1-e&q=%22Cannot+perform+an+interactive+login+from+a+non+TTY+device%22+site%3Astackoverflow.com#ip=1). Have you looked through any of those issues for solutions?

I would double-check that you are using the correct CLI version and account. If you look through the debug logs (by adding `--debug` to your command) then those may share more insight into the issue.

**Comment by paulroujansky** [comment]
Hi @tim-finnigan,

I actually did not update my CLI, it just stop working as usual at some point. Equally, no changes in my AWS account.
I have checked both GitHub and StackOverflow for some time and tried many potential fixes without any success.

I forgot to mention that I used Powershell but it worked in the past. I tried WSL but same output.

Adding `--debug` returns the following output:
```
2023-09-28 09:22:22,298 - MainThread - awscli.clidriver - DEBUG - CLI version: aws-cli/2.13.21 Python/3.11.5 Windows/10 exe/AMD64
```
